### PR TITLE
recover from idx_eth_tx_attempts_hash constraint violation

### PR DESCRIPTION
**Pull request by reductionista**
When a replay is triggered with "abandon pending transactions", the unconfirmed transactions all get marked in eth_txes as state=fatal_error, error=abandoned
However, the `idx_eth_tx_attempts_hash` db constraint enforces there can never be 2 tx attempts with exactly the same hash.  This leads to a bug where it becomes impossible to resubmit the same transaction with the same hash.

This PR adds detection of this case, and recovers by removing the abandoned attempt from `eth_tx_attempts` and replacing it with the new one.  If the corresponding entry in eth_txes is not marked as abandoned, we leave the existing attempt as-is and
return the constraint violation error as usual.

### CODE REVIEW

File: core/chains/evm/txmgr/txmgr.go [hunk 1]
```diff
@@ -53,6 +53,17 @@ type TxManager[
...
+type TestTxManager[
+	CHAIN_ID txmgrtypes.ID,
+	HEAD txmgrtypes.Head,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+] interface {
+	TxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH]
+	Abandon(addr ADDR) (err error)
+}
```
Comment (jmank88) [inline core/chains/evm/txmgr/txmgr.go:65]: Unused?
```suggestion
```

File: core/chains/evm/txmgr/txmgr.go [hunk 2]
```diff
@@ -222,10 +233,10 @@ func (b *Txm[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Reset(call
 	return err
 }
 
-// abandon, scoped to the key of this txm:
+// Abandon, scoped to the key of this txm:
```
Comment (jmank88) [inline core/chains/evm/txmgr/txmgr.go:236]: Is this only exported for test access? If so, could use a test file wrapper instead:
```go
// helpers_test.go

func (b *Txm[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Abandon(addr ADDR) (err error) {
	return b.abandon(addr)
}
```

**Review by samsondav — DISMISSED** [review]

File: core/chains/evm/txmgr/txmgr.go (hunk 1 above)
Comment (reductionista) [reply core/chains/evm/txmgr/txmgr.go]: Thought it was going to be more useful... removed that and add the helper function instead.

File: core/chains/evm/txmgr/evm_tx_store.go [hunk 3]
```diff
@@ -1281,6 +1280,45 @@ func (o *evmTxStore) UpdateEthTxAttemptInProgressToBroadcast(etx *EvmTx, attempt
...
+// case, we uphold the constraint, leaving the original tx attempt as-is and returning the constraint violation error.
+//
+// Note:  the record of the original abandoned transaction will remain in eth_txes, only the attempt is replaced.  (Any receipt
+// associated with the abandoned attempt would also be lost, although this shouldn't happen since only unconfirmed transactions
+// can be abandoned.)
+func recoverFromTxAttemptsHashConflict(attempt *EvmTxAttempt, tx pg.Queryer, dupErr error, query string, args ...interface{}) (err error) {
+	var duplicateTx DbEthTx
+	defer func() {
+		if err != nil {
+			err = multierr.Append(dupErr, err)
```
Comment (jmank88) [inline core/chains/evm/txmgr/evm_tx_store.go:1295]: nit/ std lib is sufficient now
```suggestion
			err = errors.Join(dupErr, err)
```

File: core/chains/evm/txmgr/evm_tx_store_test.go [hunk 4]
```diff
@@ -1240,6 +1241,61 @@ func TestORM_UpdateEthTxUnstartedToInProgress(t *testing.T) {
...
+					ReaperInterval:       zero,
+					ReaperThreshold:      zero,
+					ResendAfterThreshold: zero,
+				},
+			})
+			c.EVM[0] = &v2.EVMConfig{
+				ChainID: id,
+				Enabled: ptr(true),
+				Chain:   ccfg,
+			}
```
Comment (jmank88) [inline core/chains/evm/txmgr/evm_tx_store_test.go:1270]: Do you need to recreate all these settings? Or would it be sufficient to zero out the relevant fields?
```suggestion
			c.EVM[0].ReaperInterval  = zero
			...
```

File: core/chains/evm/txmgr/txmgr.go (hunk 2 above)
Comment (jmank88) [reply core/chains/evm/txmgr/txmgr.go:225]:
```suggestion
// abandon, scoped to the key of this txm:
```

**Review by jmank88 — APPROVED** [review]

**Pull request merged**